src/AlternateLife.RageMP.Net/Helpers/EventDispatcher/NativeAsyncEventDispatcher.cs

```diff
@@ -0,0 +1,56 @@
+using System.Linq;
+using System.Runtime.InteropServices;
+using AlternateLife.RageMP.Net.Enums;
+using AlternateLife.RageMP.Net.Native;
+using AlternateLife.RageMP.Net.Scripting;
+
+namespace AlternateLife.RageMP.Net.Helpers.EventDispatcher
+{
+    internal class NativeAsyncEventDispatcher<TNative, TEvent> : AsyncEventDispatcher<TEvent> where TEvent : System.EventArgs
+    {
+        private readonly EventType _type;
+        private readonly TNative _nativeCallback;
+        private readonly bool _forceRegistration;
+
+        internal NativeAsyncEventDispatcher(Plugin plugin, EventType type, TNative nativeCallback, bool forceRegistration = false) : base(plugin, type.ToString())
+        {
+            _type = type;
+            _nativeCallback = nativeCallback;
+            _forceRegistration = forceRegistration;
+
+            if (_forceRegistration)
+            {
+                Rage.Events.RegisterEventHandler((int) _type, Marshal.GetFunctionPointerForDelegate(_nativeCallback));
+            }
+        }
+
+        public override bool Subscribe(AsyncEventHandler<TEvent> callback)
+        {
+            var wasEmpty = _subscriptions.Any() == false;
+            var wasAdded = base.Subscribe(callback);
+
+            if (_forceRegistration || wasAdded == false || wasEmpty == false)
+            {
+                return true;
```

Comment:
I don't like the idea, that this method always returns true. If the subscription was not added, because there is already one registered, shouldn't it return false?